drivers/serial/serial.c

```diff
@@ -1683,14 +1659,14 @@ int uart_register(FAR const char *path, FAR uart_dev_t *dev)
     }
 #endif
 
-  /* Initialize semaphores */
+  /* Initialize mutex & semaphores */
 
-  nxsem_init(&dev->xmit.sem, 0, 1);
-  nxsem_init(&dev->recv.sem, 0, 1);
-  nxsem_init(&dev->closesem, 0, 1);
+  nxmutex_init(&dev->xmit.lock);
+  nxmutex_init(&dev->recv.lock);
+  nxmutex_init(&dev->closelock);
   nxsem_init(&dev->xmitsem,  0, 0);
   nxsem_init(&dev->recvsem,  0, 0);
-  nxsem_init(&dev->pollsem,  0, 1);
+  nxmutex_init(&dev->polllock);
```

Comment:
ditto. either keep like it is or call `nxmutex_destroy` + `nxsem_destroy` if `register_driver` fails

Reply:
keep